### ISSUE TITLE
Boost client does not take the file size into account when calculating final price for a deal with flag --storage-price-per-epoch

### ISSUE THREAD

**Issue by LexLuthr** [filecoin-project/boost#428]
### Checklist

- [X] This is **not** a question or a support request. If you have any boost related questions, please ask in the [discussion forum](https://github.com/filecoin-project/boost/discussions).
- [X] This is **not** a new feature or enhancement request. If it is, please open a [new idea discussion](https://github.com/filecoin-project/boost/discussions/new?category=ideas) instead. New feature and enhancement requests would be entertained by the boost team after a thorough discussion only.
- [X] I **have** searched on the [issue tracker](https://github.com/filecoin-project/boost/issues) and the [discussion forum](https://github.com/filecoin-project/boost/discussions), and there is no existing related issue or discussion.
- [X] I am running the [`Latest release`](https://github.com/filecoin-project/boost/releases), or the most recent RC(release canadiate) for the upcoming release or the dev branch(master), or have an issue updating to any of these.
- [X] I did not make any code changes to boost.

### Boost component

- [ ] boost daemon - storage providers
- [X] boost client
- [ ] boost UI
- [ ] boost data-transfer
- [ ] boost index-provider
- [ ] Other

### Boost Version

```text
boost 0.0.1+git.d5284dc
```

### Describe the Bug

The flag --storage-price-per-epoch does not calculate per GiB when proposing a deal. Example:

```
boost --vv deal --verified=false --provider=f0678914 --http-url=https://anton-public-bucket-boost.s3.eu-central-1.amazonaws.com/randfile_4GB_0.car --commp=baga6ea4seaqaprayexhzm7zucdwiphycnt3gqbzxhb36afh54othjdrl7x2iioa --car-size=2000177948 --piece-size=2147483648 --payload-cid=bafyaa6qsgafcmalqudsaeien3nx6fkbwsh5tdk6qsqmwaqyf4qb6zufiqwygz6j5j375v5qknejaagelucbyabasgafcmalqudsaeig45oohptp2y3q24zcazfg77felm6yeimdmqc6scpeeo4n2dunybqjaage3r7m3saykcqeaegeavdllsbzaqcaibaaeecakrvvzam --storage-price-per-epoch 10000000000

2022-04-09T07:22:54.236Z	DEBUG	boost	boost/deal_cmd.go:146	selected wallet	{"wallet": "f3wkp4blevjsrtbc6vwgjf2sedzjwsqmj3wsh4uexbp4k7dggs72kbvuv7xivsnz7cnmfazpmqp3qmchmzms6a"}
2022-04-09T07:22:54.237Z	DEBUG	boost	boost/deal_cmd.go:158	found storage provider	{"id": "12D3KooWAJwyjLgKAARsVqFv61PCWRQSajtfajywj6rmaBMYAGoH", "multiaddrs": ["/ip4/199.73.55.234/tcp/10221"], "addr": "f0678914"}
2022-04-09T07:22:54.728Z	DEBUG	boost	boost/deal_cmd.go:239	current block height	{"number": "1706085"}
2022-04-09T07:22:54.730Z	DEBUG	boost	boost/deal_cmd.go:258	about to submit deal proposal	{"uuid": "b54c324e-f3e5-41a6-80a7-79f6d5efffe8"}
2022-04-09T07:22:54.930Z	FATAL	boost	boost/main.go:42	deal proposal rejected: failed validation: validateAsk failed: storage price per epoch less than asking price: 10000000000 < 20000000000
```

Here, client has set the limit as 10000000000 per GiB per Epoch for a storage deal. Now, ideally when I am proposing a deal, I will get this number from the client and add it to the boost deal command. The boost client should calculate the final amount based on the file size and --storage-price-per-epoch for the proposal. Otherwise, this quickly becomes a rather tedious exercise of calculating the deal price for each proposal manually. 

### Logging Information

```text
No logging required
```

### Repo Steps

1. Run '...'
2. Do '...'
3. See error '...'
...

**Comment by dirkmc** [comment]
Related: https://github.com/filecoin-project/boost/issues/386

**Comment by jacobheun** [comment]
Fixed via #512 and #513
